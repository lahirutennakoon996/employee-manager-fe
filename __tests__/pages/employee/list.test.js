import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';

import List from "@/pages/employee/list";
import mockStore from "../../../__mocks__/react-redux.mock";

describe('Employee list', () => {
  const initialState = {
    employees: {
      allEmployees: [],
    }
  };
  const store = mockStore(initialState);
  let component;

  beforeEach(() => {
    const componentRendered = render(<Provider store={store}><List /></Provider>);
    component = componentRendered.container;
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders the Add button', () => {
    const addEmployeeBtn = screen.getByRole('button', {
      name: /Add Employee/i,
    })

    expect(addEmployeeBtn).toBeInTheDocument();
    expect(addEmployeeBtn).not.toBeDisabled();
  })

  it('render the Toggle View button', async () => {
    const toggle = screen.getByRole('button', {
      name: /toggle view/i,
    });

    expect(toggle).toBeInTheDocument();
    expect(toggle).not.toBeDisabled();

    // Check for grid view
    const gridView = screen.getByTestId('grid-view');
    expect(gridView).toBeInTheDocument();

    // Check for table view after clicking 'toggle' button
    await userEvent.click(toggle);
    expect(gridView).not.toBeInTheDocument();
    const tableView = screen.getByTestId('table-view');
    expect(tableView).toBeInTheDocument();
  })
})