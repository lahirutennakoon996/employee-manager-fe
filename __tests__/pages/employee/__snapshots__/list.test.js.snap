// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee list should match snapshot 1`] = `
<div>
  <div
    class="container "
  >
    <div
      class="row text-end"
    >
      <div
        class="col"
      >
        <button
          class="btn btn-primary my-2"
          type="button"
        >
          Add Employee
        </button>
         
        <button
          class="btn btn-primary my-2"
          type="button"
        >
          Toggle view
        </button>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="row"
        data-testid="grid-view"
      />
    </div>
  </div>
</div>
`;
